frontend/src/app/(pages)/questions/[id]/page.tsx

```diff
@@ -0,0 +1,18 @@
+"use client"
+import Question from "@/types/question";
+import { getQuestionById } from "../api";
+import { useParams } from "next/navigation";
+
+export default async function QuestionDetailPage() {
```

Comment:
I think another way to get the dynamic parameter is that outside this function, you create an interface outside of this function, and then you can access the params without the need of using the `useParams` hook. With this approach, the benefit is that the page can be a server component. You can refer to the screenshot I attached for reference.
![image](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g05/assets/69421979/2dd0b56d-f21e-4e0f-bbc4-62144a949641)